upf/action.py

```diff
@@ -163,3 +159,6 @@ def add_decrease_effect(self, fluent: Union[FNode, 'upf.Fluent'],
         if not self._env.type_checker.is_compatible_type(fluent_exp, value_exp):
             raise UPFTypeError('Action effect has not compatible types!')
         self._effects.append(Effect(fluent_exp, value_exp, condition_exp, kind = DECREASE))
+
+    def add_effect_class(self, effect: Effect):
```

Comment:
I would rename this method `add_effect` and the original one in `add_assign_effect`